File: edukate-checker/src/main/java/io/github/sanyavertolet/edukate/checker/configs/RabbitConfig.java
package io.github.sanyavertolet.edukate.checker.configs;

import io.github.sanyavertolet.edukate.messaging.RabbitTopology;
import org.springframework.amqp.core.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitConfig {

    @Bean
    public Declarables scheduleTopology(TopicExchange edukateExchange) {
        Queue scheduleQueue = QueueBuilder.durable(RabbitTopology.Q.SCHEDULE_CHECKER).build();
        Binding binding = BindingBuilder.bind(scheduleQueue).to(edukateExchange).with(RabbitTopology.Rk.SCHEDULE);
        return new Declarables(scheduleQueue, binding);
    }
}

File: edukate-checker/src/main/kotlin/io/github/sanyavertolet/edukate/checker/configs/RabbitConfig.kt
package io.github.sanyavertolet.edukate.checker.configs

import io.github.sanyavertolet.edukate.messaging.RabbitTopology
import org.springframework.amqp.core.BindingBuilder
import org.springframework.amqp.core.Declarables
import org.springframework.amqp.core.QueueBuilder
import org.springframework.amqp.core.TopicExchange
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitConfig {
    @Bean
    fun scheduleTopology(edukateExchange: TopicExchange): Declarables {
        val scheduleQueue = QueueBuilder.durable(RabbitTopology.Q.SCHEDULE_CHECKER).build()
        val binding = BindingBuilder.bind(scheduleQueue).to(edukateExchange).with(RabbitTopology.Rk.SCHEDULE)
        return Declarables(scheduleQueue, binding)
    }
}
